libpromises/eval_context.c

```diff
@@ -129,11 +129,24 @@ struct EvalContext_
 
     /* new package promise evaluation context */
     PackagePromiseContext *package_promise_context;
+    
+    /* select_end_match_eof context*/
+    bool select_end_match_eof;
 
     /* List if all classes set during policy evaluation */
     StringSet *all_classes;
 };
 
+bool GetSelectEndMatchEof(const EvalContext *ctx)
+{
+    return ctx->select_end_match_eof;
+}
+
+void SetSelectEndMatchEof(EvalContext *ctx, bool value)
```

Comment:
Most functions regarding `EvalContext` are prefixed with `EvalContext`. We should probably do that here as well.